### PR TITLE
hss: init at 1.0.1

### PR DESCRIPTION
**Pull request by nixy**
###### Motivation for this change

Adds hss, a simple SSH helper written in ruby to allow shorthand for a fleet of hosts.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/networking/hss/default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{ lib, bundlerEnv, ruby }:
+
+bundlerEnv rec {
+  name = "hss-${version}";
+  version = "1.0.1";
+
+  inherit ruby;
+  gemfile  = ./Gemfile;
+  lockfile = ./Gemfile.lock;
+  gemset   = ./gemset.nix;
```
Comment (Mic92) [inline pkgs/tools/networking/hss/default.nix]: Just use `gemdir = ./.;` instead of the three lines above.

Comment (nixy) [inline pkgs/tools/networking/hss/default.nix]: Fair enough, this seems simpler and works just the same. Thanks for the feedback :)

**Comment by Mic92** [comment]
Would it not also useful to override https://github.com/akerl/hss/blob/master/bin/hss#L21 with a reference to the `ssh` executable?

**Comment by nixy** [comment]
Yeah, now that I think about it that path should absolutely be replaced. I was a bit tired when I put this together, so I missed that.

What would be the best way to replace that line? I thought about using sed on the output in postInstall, but the path is very long and includes version numbers for ruby. It seems ill advised to do that. What phase would patching that make sense at?

**Comment by Mic92** [comment]
use `substituteInPlace` in `postPatch` phase.

**Comment by nixy** [comment]
I tried a couple of things, but it looks like bundlerEnv doesn't support most of the normal phases and overrides. The only extra commands you can run are postBuild, at which point the files are already in place behind the ruby versions. So any code I place in a postInstall/postPatch doesn't get executed at all.

Replacing ssh here doesn't seem to be as easy I first thought. I can see a few examples in nixpkgs where bundlerEnv is tied into a stdenv.mkderivation, so I think I might refer to them to see what they did.

**Pull request merged**